### PR TITLE
Added RKN methods

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I started a PR for Verner Methods #2274

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas Can you please review this PR to see if there are an changes needed here or we are ready to merge this ??

File: lib/OrdinaryDiffEqRKN/src/OrdinaryDiffEqRKN.jl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module OrdinaryDiffEqRKN
+
+import OrdinaryDiffEq: alg_order, calculate_residuals!,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqRKN/src/OrdinaryDiffEqRKN.jl:5]:
```suggestion
                       calculate_residuals, alg_extrapolates,
```

File: lib/OrdinaryDiffEqRKN/src/OrdinaryDiffEqRKN.jl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import OrdinaryDiffEq: alg_order, calculate_residuals!,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, alg_extrapolates,
+                       OrdinaryDiffEqAlgorithm,
+                       OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+                       OrdinaryDiffEqAdaptivePartitionedAlgorithm,
+                       OrdinaryDiffEqPartitionedAlgorithm,
+                       OrdinaryDiffEqAdaptiveAlgorithm, CompiledFloats, uses_uprev,
+                       alg_cache, _vec, _reshape, @cache, isfsal, full_cache,
+                       constvalue, _unwrap_val, du_alias_or_new,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqRKN/src/OrdinaryDiffEqRKN.jl:12]:
```suggestion
                       constvalue, _unwrap_val, du_alias_or_new, _ode_interpolant,
```

**Pull request merged**